packages/VersionBonding/PhpVersionedFilter.php

```diff
@@ -25,6 +28,15 @@ public function filter(iterable $rectors): array
 
         $activeRectors = [];
         foreach ($rectors as $rector) {
+            if ($rector instanceof RelatedPolyfillInterface) {
+                $polyfillPackageNames = $this->polyfillPackagesProvider->provide();
```

Comment:
this can be moved before loop

Reply:
alternatively, init before loop:

```php
  $polyfillPackageNames = null;
```

and use coalesce on loop:

```php
  $polyfillPackageNames ??= this->polyfillPackagesProvider->provide();
```

That cache is then no longer needed, as cached on loop itself to use isset data when available.